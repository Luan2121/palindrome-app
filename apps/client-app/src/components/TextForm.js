import { useState } from 'react';
import { Row, Form, Button, Col } from 'react-bootstrap';

const TextForm = ({ 
    onSubmit
}) => {

    const [text,setText] = useState('');

    const handleChange = (event) => {
        setText(event.target.value);
    }

    const handleSubmit = (event) => {
        if(typeof onSubmit === 'function'){
            onSubmit(event,{text});
        }
        setText("");
    }

    return (
        <Row md = "6" className = "justify-content-center w-100">
            <Col as =  {Form} md = "6" className = "d-flex" onSubmit = {handleSubmit}>
                <Form.Group className = "me-3 w-100">
                    <Form.Control 
                        placeholder = "Insert Text" 
                        value = {text} 
                        onChange = {handleChange} 
                    />
                </Form.Group>
                <Button type = "submit" disabled = {!text}>
                    Send
                </Button>
            </Col>
        </Row>
    )
}

export default TextForm;