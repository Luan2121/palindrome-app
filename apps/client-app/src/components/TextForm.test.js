import { render, fireEvent } from '@testing-library/react';
import TextForm from './TextForm'

describe('Form Component', () => {

    test('should render correctly', () => {
        const { asFragment } = render(<TextForm />);
        expect( asFragment() ).toMatchSnapshot();
    })

    test('it should have button disabled if input has no text', () => {
        const { getByRole } = render(<TextForm />)
        const button = getByRole('button');
        expect(button).toBeDisabled();
    })

    test('it should have button enabled if input has text', () => {
        const { getByRole, getByPlaceholderText } = render(<TextForm />);
        const input = getByPlaceholderText('Insert Text')
        const button = getByRole('button');

        fireEvent.change(input, { target: { value: 'Test' } });
        
        expect(button).not.toBeDisabled();
    })
    
})