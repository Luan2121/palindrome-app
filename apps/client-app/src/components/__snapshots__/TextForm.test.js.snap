// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Component should render correctly 1`] = `
<DocumentFragment>
  <div
    class="justify-content-center w-100 row row-cols-md-6"
  >
    <form
      class="d-flex col-md-6"
    >
      <div
        class="me-3 w-100"
      >
        <input
          class="form-control"
          placeholder="Insert Text"
          value=""
        />
      </div>
      <button
        class="btn btn-primary"
        disabled=""
        type="submit"
      >
        Send
      </button>
    </form>
  </div>
</DocumentFragment>
`;
